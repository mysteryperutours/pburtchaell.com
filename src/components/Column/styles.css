.column {
  position: relative;
  flex: 0 0 auto;
  flex-grow: 1;
  flex-basis: 0;
  flex-direction: column;
  box-sizing: border-box;
}

/*
 * Only add a gutter to rows that have padding.
 */
.row:not(.padding-size-none) > .column {
  padding-right: 2rem;
  padding-left: 2rem;
}

.row.padding-size-none > .column:last-of-type {
  padding-right: 2rem;
}

@media only screen and (min-width: 600px) {
  .column-flex-order-large-0 {
    order: 0;
  }

  .column-flex-order-large-1 {
    order: 1;
  }

  .column-flex-order-large-2 {
    order: 2;
  }

  .column-size-large-3 {
    max-width: calc((3 * 100%) / 12);
    flex-basis: calc((3 * 100%) / 12);
  }

  .column-size-large-6 {
    max-width: calc((6 * 100%) / 12);
    flex-basis: calc((6 * 100%) / 12);
  }

  .column-size-large-6:only-child {
    margin-left: calc((3 * 100%) / 12);
    max-width: calc((6 * 100%) / 12);
    flex-basis: calc((6 * 100%) / 12);
  }

  .column-size-large-9 {
    max-width: calc((9 * 100%) / 12);
    flex-basis: calc((9 * 100%) / 12);
  }

  .column-size-large-12 {
    max-width: calc((12 * 100%) / 12);
    flex-basis: calc((12 * 100%) / 12);
  }
}

@media only screen and (max-width: 600px) {
  .column-flex-order-small-0 {
    order: 0;
  }

  .column-flex-order-small-1 {
    order: 1;
  }

  .column-flex-order-small-2 {
    order: 2;
  }

  .column-hide-on-small {
    display: none;
  }

  .column-size-small-4 {
    max-width: calc((4 * 100%) / 12);
    flex-basis: calc((4 * 100%) / 12);
  }

  .column-size-small-6 {
    max-width: calc((6 * 100%) / 12);
    flex-basis: calc((6 * 100%) / 12);
  }

  .column-size-small-8 {
    max-width: calc((8 * 100%) / 12);
    flex-basis: calc((8 * 100%) / 12);
  }

  .column-size-small-12 {
    max-width: calc((12 * 100%) / 12);
    flex-basis: calc((12 * 100%) / 12);
  }
}
