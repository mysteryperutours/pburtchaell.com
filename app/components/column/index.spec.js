import React from 'react';
import Column from './index';

describe(Column.name, () => {
  it('should render 4 column component', () => {
    expect(shallow(
      <Column largeSize="4" smallSize="4">
        <div />
      </Column>
    )).toMatchSnapshot();
  });

  it('should render 6 column component', () => {
    expect(shallow(
      <Column largeSize="6" smallSize="6">
        <div />
      </Column>
    )).toMatchSnapshot();
  });

  it('should render 8 column component', () => {
    expect(shallow(
      <Column largeSize="8" smallSize="8">
        <div />
      </Column>
    )).toMatchSnapshot();
  });

  it('should render 12 column component', () => {
    expect(shallow(
      <Column>
        <div />
      </Column>
    )).toMatchSnapshot();
  });

  it('should render component that hides on small screens', () => {
    expect(shallow(
      <Column hideOnSmall={true}>
        <div />
      </Column>
    )).toMatchSnapshot();
  });

  it('should render component with unique element type', () => {
    expect(shallow(
      <Column elementType="section">
        <div />
      </Column>
    )).toMatchSnapshot();
  });
});
