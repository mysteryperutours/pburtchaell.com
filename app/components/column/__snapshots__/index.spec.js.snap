// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column should render 4 column component 1`] = `
<div
  className="column column-large-4 column-small-4"
>
  <div />
</div>
`;

exports[`Column should render 6 column component 1`] = `
<div
  className="column column-large-6 column-small-6"
>
  <div />
</div>
`;

exports[`Column should render 8 column component 1`] = `
<div
  className="column column-large-8 column-small-8"
>
  <div />
</div>
`;

exports[`Column should render 12 column component 1`] = `
<div
  className="column column-large-12 column-small-12"
>
  <div />
</div>
`;

exports[`Column should render component that hides on small screens 1`] = `
<div
  className="column column-large-12 column-small-12 hide-on-small"
>
  <div />
</div>
`;

exports[`Column should render component with unique element type 1`] = `
<section
  className="column column-large-12 column-small-12"
>
  <div />
</section>
`;
